src/script/miniscript.h

```diff
@@ -746,6 +762,30 @@ struct Node {
         return {{}, {}};
     }
 
+    //! Returns true if the node contains at least one duplicate key.
+    bool ContainsDuplicateKey() const {
+        auto upfn = [this](const Node& node, Span<std::set<Key>> subs) -> std::optional<std::set<Key>> {
+            if (&node != this && node.duplicate_key) return {};
```

Comment:
I am confused by why we need the node = this here. Should upfn always not operate on this node?

Reply:
`this->duplicate_key` isn't yet initialized when `ContainsDuplicateKey` is being called, so we need to skip inspecting it.